pkg/api/defaults-apiserver.go

```diff
@@ -164,6 +164,14 @@ func (cs *ContainerService) setAPIServerConfig() {
 	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.20.0-alpha.1") && o.KubernetesConfig.APIServerConfig["--service-account-issuer"] == "kubernetes.default.svc" {
 		o.KubernetesConfig.APIServerConfig["--service-account-issuer"] = "https://kubernetes.default.svc.cluster.local"
 	}
+
+	invalidFeatureGates := []string{}
+	// Remove --feature-gate VolumeSnapshotDataSource starting with 1.22
+	// Reference: https://github.com/kubernetes/kubernetes/pull/101531
+	if common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.22.0-alpha.1") {
+		invalidFeatureGates = append(invalidFeatureGates, "VolumeSnapshotDataSource")
+	}
+	removeInvalidFeatureGates(o.KubernetesConfig.APIServerConfig, invalidFeatureGates)
```

Comment:
why are we doing this differently than what's already in place for other flags, ie. https://github.com/Azure/aks-engine/pull/4903/files#diff-f3e26129863d5da8cf5516c2d7abf5d9fe043bbb8119e076fd96b55ec2032414L151 ?

Reply:
I see, we're removing a specific feature gate within the flag, makes sense